python/tests/parsers/test_main.py

```diff
@@ -0,0 +1,163 @@
+
+    def setUp(self):
+        """This function tests sets up test database"""
+        db.create_all()
+        self.company = Company(cik_no="0001171592", company_name="Cool Industries", filing_count=0)
+        self.edgar_filing = EdgarFiling(accession_no="0001420506-21-000830", cik_no="0001171592",
+                                        filing_date=datetime.fromisoformat("1999-09-01"))
+        self.data_13f_table = Data13f(equity_holdings_id="67896567",
+                                      accession_no='0001420506-21-000830',
+                                      cik_no='56464565767',
+                                      name_of_issuer='Agilent Technologies',
+                                      title_of_class='COM',
+                                      cusip='00846U101',
+                                      value='22967078',
+                                      ssh_prnamt='180644',
+                                      ssh_prnamt_type='None',
+                                      put_call='None',
+                                      investment_discretion='SOLE',
+                                      other_manager='None',
+                                      voting_authority_sole='22967078',
+                                      voting_authority_shared='0',
+                                      voting_authority_none='0')
+        db.session.merge(self.company)
+        db.session.merge(self.edgar_filing)
+        db.session.merge(self.data_13f_table)
+        db.session.commit()
+
+    def tearDown(self):
+        """This function tests tears down test database"""
+        db.session.remove()
+        db.drop_all()
+
+
+def test_create_url_list(current_events_text):
+    """This function tests test_parse_13f_filing_detail_urls"""
+    with patch('requests.get') as mock_function:
+        mock_function.return_value = MagicMock(text=current_events_text)
+        fake_url = ""
+
+        actual = create_url_list(fake_url)
+
+        assert actual == [
+            'https://www.sec.gov/Archives/edgar/data/1850858/0001850858-21-000001-index.html',
+            'https://www.sec.gov/Archives/edgar/data/1852858/0001852858-21-000001-index.html',
+            'https://www.sec.gov/Archives/edgar/data/1835714/0001085146-21-001095-index.html',
+            'https://www.sec.gov/Archives/edgar/data/1446194/0001011712-21-000002-index.html']
+
+
+class FlaskSQLAlchemyTest(FlaskSqlAlchemyTestConfiguration):
+    """This class runs SQL ALchemy Tests"""
+
+    def test_update_filing_counts(self):
+        """This function tests update_filing_counts"""
+        cik_no = '0001171592'
+
+        update_filing_counts([cik_no])
+
+        assert Company.query.filter_by(cik_no=cik_no).first().filing_count == 1
+
+    def test_send_data_to_db_savesCompanyInDb(self):
+        """This function tests send_data_to_db_savesCompanyInDb"""
+        self.company_2 = Company(cik_no="000984343", company_name="True Blue", filing_count=0)
```

Comment:
no need to make these variables fields of the class. Just leave them as local variables.